packages/backend/src/open_payments/auth/grant.ts

```diff
@@ -66,11 +69,13 @@ export class Grant {
     this.active = options.active
     this.grant = options.grant
     this.access = options.access || []
+    this.clientId = options.clientId
   }
 
   public readonly active: boolean
   public readonly grant: string
   public readonly access: GrantAccess[]
+  public readonly clientId: string
 
   public includesAccess({
```

Comment:
[`createAuthMiddleware`](https://github.com/interledger/rafiki/blob/68115bb912ed75bb8d72bb542641df9f5dcff1d7/packages/backend/src/open_payments/auth/middleware.ts#L4-L30) is only ever being called with `AccessAction.Read`/`AccessAction.List` and never `AccessAction.ReadAll`/`AccessAction.ListAll`.

Maybe `includesAccess` could be updated to return true for an `action` of `AccessAction.Read`/`AccessAction.List` (specified by the auth middleware) if the grant access includes `AccessAction.ReadAll`/`AccessAction.ListAll`

Reply:
Does that depend on 
- https://github.com/interledger/open-payments/issues/178

Reply:
I don't think so.
The problem is that a get or list request will currently be rejected by the auth middleware if the grant has `read-all` or `list-all` (and no `read` or `list`)